content/en/blog/attribute-autocomplete-enhancement/index.md

```diff
@@ -0,0 +1,306 @@
+
+The distance is based on the likelyhood of the change. For the edit distance for update, if the attribute name and type are unchanged, then it is most likely that the attribute itself is unchanged, thus no distance. If either attribute name or type is changed, then it has distance value 1. If both the attribute name and type are changed, it is considered unlikely, thus having distance value 2.
+
+| Attribute Name | Attribute Type | Distance |
+|--|--|--|
+| Unchanged | Unchanged | 0 |
+| Changed | Unchanged | 1 |
+| Unchanged | Changed | 1 |
+| Changed | Changed | 2 |
+
+Table 3, Attribute edit distance for update
+
+With this algorithm, for the old and new input schema example, we can say that `Column C` is believed to be renamed to `Column D`, instead of that `Column C` is deleted and `Column D` is a new attribute, because the former case has distance value 1 while the later case has distance value 2 (1 deletion+ 1 insertion). Also, `Column B` is deleted instead of renamed and type casted to `Column E`, because deletion only has distance 1, but the rename and type casted is considered an update of distance 2.
+
+#### Correct attribute selection
+
+In every operator involving input schema changes, after the determination, we update the attribute selection in the operators recursively based on the determination.
+
+The property update is conducted in a recursive way because the object of opeartor properties can be nested with deep property. All property values are recursively traversed to find the property with value of old attribute selection and update with new attribute name.
+
+### Limitations
+
+The implementation of the auto attribute correction cannot always find out the correct attribute changes because of the attribute edit distance algorithm. We should notice due to the limited information that an operator knows about its input schema, it can only "guess" what changes happened, but cannot confidently determine the changes. While we find out the most likely changes using our edit distance algorithm, the method shall not be considered 100% correct. Moreover, in some cases, two possible changes may be the same edit distance. In the Table 4 and 5 example, it is possible that `Column A` is renamed to `Column C` and `Column B` is renamed to `Column D`, meanwhile it is also possible that `Column A` is renamed to `Column D` and `Column B` is renamed to `Column C`. Both possibilities have distance 2. In this case, our edit distance algorithm may take either result.
+
+|Attribute Name|Attribute Type|
+|--|--|
+|Column A|integer|
+|Column B|integer|
+
+Table 4, Example 2 - old input schema
+
+|Attribute Name|Attribute Type|
+|--|--|
+|Column C|integer|
+|Column D|integer|
+
+Table 5, Example 2 - new input schema
+
+## Attribute Type Suggestion
```

Comment:
I also feel `suggestion` is not a good terminology here. by `suggestion` I assume you can suggest and autocomplete an attribute property based on some information (e.g., another attribute property). But I think this is not what you meant. 

How about `Attribute Type Validation` or `Attribute Type Static Check`?